### ISSUE TITLE
Pick does NOT work with SubSchemas

### ISSUE THREAD

**Issue by thebarty** [aldeed/meteor-simple-schema#514]
Hi guys,

I think I found a bug: when **picking** a field with **type=subschema**, the returned schema does NOT contain the subschema.

**This is the example code - am I missing something here?**

```
SubSchema = new SimpleSchema({
  name: {
    type: String,
  },
});

MainSchema = new SimpleSchema({
  // ..
  subschema: {
    type: SubSchema,
  },
  // ..
});

// BUG BUG BUG
// Pick does NOT work as excpected
// .. this only returns a FLAT object, with NO "name"-field attached
AnotherSchema = MainSchema.pick(['subschema']);  // Note: .pick(['subschema', 'subschema.$']); does NOT help
```

**Comment by pmoons** [comment]
I am experiencing this issue as well.

**Comment by egoldblum** [comment]
+1 this burned me too.  Very misleading behavior when trying to compose complex schemas together.  Would prefer no API to a misleading one.

**Comment by aldeed** [comment]
I believe I fixed this in 2.0. If not, please submit a new issue in the node repo: https://github.com/aldeed/node-simple-schema
